### PR TITLE
run: fix hang when using run

### PR DESCRIPTION
**Pull request by giuseppe**
make the stdin pipe not blocking, so that it won't hang if the other
end is not reading from it.

Closes: https://github.com/containers/buildah/issues/1664

Signed-off-by: Giuseppe Scrivano <gscrivan@redhat.com>

### CODE REVIEW

**Comment by TomSweeneyRedHat** [comment]
@giuseppe 
Looks like you need to make a function somewhere to avoid gofmt cycle issues:
```
# Run gofmt on version 1.11 and higher
run_linux.go:1052::warning: cyclomatic complexity 46 of function runCopyStdio() is high (> 45) (gocyclo)
Makefile:66: recipe for target 'validate' failed
make: *** [validate] Error 1
```

**Comment by rhatdan** [comment]
LGTM

**Comment by TomSweeneyRedHat** [comment]
looks like a network flake killed it, will spin again.

File: run_linux.go [hunk 1]
```diff
@@ -1049,6 +1049,18 @@ func runConfigureNetwork(isolation Isolation, options RunOptions, configureNetwo
 	return teardown, nil
 }
 
+func setNonblock(fd int, description string, blocking bool) error {
+	err := unix.SetNonblock(fd, true)
```
Comment (mheon) [inline run_linux.go]: Is this right? Shouldn't this be `fd, blocking` instead of unconditional true?

Comment (giuseppe) [reply run_linux.go]: argh, sure.  Surprised the tests passed

**Comment by rhatdan** [comment]
@rh-atomic-bot r+

**Comment by giuseppe** [comment]
@rh-atomic-bot retry

**Comment by TomSweeneyRedHat** [comment]
I've rekicked the two failing travis tests. It looks like they're completing or at least mostly completing but running out of time?

**Comment by giuseppe** [comment]
it seems the same issue is present in chroot, I am looking at it right now

**Comment by rhatdan** [comment]
@rh-atomic-bot r+

**Pull request closed without merging**